src/dialog/Dialog.ts

```diff
@@ -81,11 +82,9 @@ export class DialogBase<P extends DialogProperties = DialogProperties> extends T
 		document.addEventListener('keyup', this._onKeyUp);
 	}
 
-	protected onElementCreated(element: HTMLElement, key: string) {
-		if (key === 'main') {
-			const { open = false } = this.properties;
-			open && element.focus();
-		}
+	protected onAttach() {
+		const { open = false } = this.properties;
+		open && this.meta(Focus).set('main');
```

Comment:
this isn't working, `onAttach` is only called when the widget as a whole is added to the DOM, not when the dialog element is created and attached.

Reply:
I forgot to add functional focus tests when I added focus handling to the dialog, but those would've been useful 😝

Reply:
Ha that would have let me know it wasn't working!

Reply:
Have moved this to be called when the `open` property is `true` and the dialog was not open previously (already had this state in the widget)